### DISCUSSION TITLE
change default search engine

### DISCUSSION THREAD

**Discussion by almqq** [brookhong/Surfingkeys#1453] [Q&A]
Hi, any idea how I change the default search engine. For example, if I press "t" and then search for something, it uses google. I'd like to change this to oneSearch
Thanks

**Comment by cutbypham** [comment]
README.md

![image](https://user-images.githubusercontent.com/40050527/119247717-5639c280-bbb6-11eb-8f94-9597d0882615.png)

**eg. I switch to default search to duckduckgo = 'd'**

![image](https://user-images.githubusercontent.com/40050527/119247750-8e410580-bbb6-11eb-99cc-77ed1e2160cd.png)

**Comment by boxallar** [comment]
Hi almqq, thanks for the advice.
I have tried putting this line of code into my settings but without any luck.
It still uses google and I get this message
`[SurfingKeys] Error found in settings: ReferenceError: mapkey is not defined`
I have tried using d, b and even the url for duckduckgo, without any luck.

Can someone please help me?

**Comment by boxallar** [comment]
Hmmm...
when I reset the settings everything goes back to normal.
When I even add a comment line with `//` then I get the same error message.

I am using Firefox in case that helps

**Comment by boxallar** [comment]
the problem is still there.  Editing settings, by adding a line of comments still gives me the error message as soon as I do anything

**Comment by Archie-2021** [comment]
Is  there a way to set up a new search engine or change what triggers them? How I can do that?

**Reply by boxallar** [reply]
Hi Archie, if you click on the surfingkeys extension icon and then on settings you can click on or off different search engines.  Make sure that advanced mode is not on to see these options.
You can by default search with og for google, ob for baidu and so on.
As far as adding a new search engine, I only know the advice above, which did not work for me.  If you find a solution, please do let me know
